src/evaluators/utility/PHAL_ComputeBasisFunctions_Def.hpp

```diff
@@ -100,12 +99,12 @@ evaluateFields(typename Traits::EvalData workset)
   typedef typename Intrepid2::CellTools<PHX::Device>   ICT;
   typedef Intrepid2::FunctionSpaceTools<PHX::Device>   IFST;
 
-  ICT::setJacobian(jacobian, refPoints, coordVec.get_view(), intrepidBasis);
+  ICT::setJacobian(jacobian, refPoints, coordVec.get_view(), intrepid2MapBasis);
```

Comment:
Is there a reason not to use the FE basis for this?

Reply:
Yes, the Jacobian is based on the mapping between the reference and the physical element, which is different in general from the finite element field defined on the element. You could have an affine mapping and then use P2 as finite element, or, like in the depth-integrated case a weird forth order basis.